### ISSUE TITLE
Creating folder in file menu

### ISSUE THREAD

**Issue by vzocca** [spyder-ide/spyder#21477]
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

### What steps reproduce the problem?

1. 
2. 
3. 

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 
* Python version: 
* Qt version: 
* PyQt version: 
* Operating System name/version: 

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by vzocca** [comment]
This is not a bug but a feature request. Currently, one can open several python files and they will be created as tags in the file menu and one can move from a file to another selecting a different tag. It would be useful to add some folders as well, similar to how one can create bookmarks in a browser. One could also have these foolders (they could be separate projects) in the file menu as tags and clicking on them would show the different folders in that project. These folders do not need to be the same as physical folders on disk, but, like bookmarks, logical ways of aggregating files for a cleaner and more organised look.

**Comment by ccordoba12** [comment]
Hey @vzocca, thanks for reporting. You said:

> This is not a bug but a feature request. Currently, one can open several python files and they will be created as tags in the file menu and one can move from a file to another selecting a different tag.

You are referring to the list of files that are shown in the `File > Open Recent` menu?

After let me know if that's correct, I'll try to answer your feature request.

**Comment by vzocca** [comment]
Thank you so much for the quick reply. (I also replied via e-mail)
I am referring to the tags on top of the editor (see screenshot, first file attached).
Each tab is a file. You can load dozens of them. Can we add the possibility to group them in a folder (that does not have to be a physical folder, more like a logical way of grouping the files by project).
I might be working on several projects at once, and each project may comprise several files. This would allow someone to easily switch between projects and find the file needed.

This is similar to how chrome handles bookmarks. My bookmarks can be grouped in folders (see I have "Jobs", I have "Research") or single bookmarks (I have "Dopamine", "IT", "UK") -see second screenshot, second file attached.

![spyder_files](https://github.com/spyder-ide/spyder/assets/6509857/a55e882a-f855-4907-a7a4-e493cd442171)

![chrome](https://github.com/spyder-ide/spyder/assets/6509857/f89d670b-4f2f-40ed-b0f1-bc55a0419f9d)

**Comment by vzocca** [comment]
Thank you so much for the quick reply.
I am referring to the tags on top of the editor (see screenshot).
Each tab is a file. You can load dozens of them. Can we add the possibility
to group them in a folder (that does not have to be a physical folder, more
like a logical way of grouping the files by project).
I might be working on several projects at once, and each project may
comprise several files. This would allow someone to easily switch between
projects and find the file needed.

This is similar to how chrome handles bookmarks. My bookmarks can be
grouped in folders (see I have "Jobs", I have "Research") or single
bookmarks (I have "Dopamine", "IT", "UK").

I hope this is clearer than my first post.
Thank you!

Valentino


[image: spyder_files.png]

[image: chrome.png]

Il giorno mar 31 ott 2023 alle ore 14:11 Carlos Cordoba <
***@***.***> ha scritto:

**Comment by ccordoba12** [comment]
> Each tab is a file. You can load dozens of them. Can we add the possibility to group them in a folder (that does not have to be a physical folder, more like a logical way of grouping the files by project).
I might be working on several projects at once, and each project may comprise several files. This would allow someone to easily switch between projects and find the file needed.

Thanks for for the clearer explanation! I understand your suggestion now and I think it'd be a very cool feature to have.

Unfortunately, we're currently understaffed and don't have the resources to implement it. But I'll leave your issue open in case the situation changes in the future.

**Comment by vzocca** [comment]
I am glad you liked my suggestion. I understand there may not be enough
bandwidth to do it now, but I am happy you will keep it as a suggestion and
it may get implemented eventually.

Il giorno mer 1 nov 2023 alle ore 19:10 Carlos Cordoba <
***@***.***> ha scritto:

**Comment by OverLordGoldDragon** [comment]
Seconding this, a nice to have.

In case my vision differs from OP's, I just want to group opened files, preferably on per-project basis (one project's groupings don't affect others').

**Comment by vzocca** [comment]
Yes, that is what I would like to have, to be able to group opened files. I hope enough people will find it useful so that it can be implemented.

**Comment by gamesbook** [comment]
I recently posted to the mailing list:

_I find that having to switch to work on different aspects (features and/or branches) for a given project becomes very tedious. What I need is a way to save different sets of tabs (open files) under different names, and then to easily switch between them from a drop-down (or look-up) list._

I support the folders idea (implemented per project), provided that the same file can appear in multiple folders.
